### PR TITLE
Remove double fetch of "OSSL_EX_DATA_GLOBAL" for global lock

### PR DESCRIPTION
**Pull request by davxy**
Fetch once and just pass the global ex_data to the "get_and_lock" static
function.

Removed a redundant null pointer check within the "get_and_lock" static
function (control already performed by the caller).

### CODE REVIEW

File: crypto/ex_data.c [hunk 1]
```diff
@@ -12,8 +12,9 @@
 
 int do_ex_data_init(OPENSSL_CTX *ctx)
 {
-    OSSL_EX_DATA_GLOBAL *global = openssl_ctx_get_ex_data_global(ctx);
+    OSSL_EX_DATA_GLOBAL *global;
```
Comment (richsalz) [inline crypto/ex_data.c]: why did you split this into two lines?

File: crypto/ex_data.c [hunk 2]
```diff
@@ -24,22 +25,21 @@ int do_ex_data_init(OPENSSL_CTX *ctx)
...
         return NULL;
     }
 
-    global = openssl_ctx_get_ex_data_global(ctx);
-    if (global == NULL || global->ex_data_lock == NULL) {
+    if (global->ex_data_lock == NULL) {
         /*
-         * If we get here, someone (who?) cleaned up the lock, so just
-         * treat it as an error.
+         * If we get here, someone cleaned up the lock, so just treat
```
Comment (richsalz) [inline crypto/ex_data.c]: why did you remove the fact that we don't know who?  It's arguably a bug/TBD/TODO to find out.

File: crypto/ex_data.c [hunk 3]
```diff
@@ -63,8 +63,9 @@ static void cleanup_cb(EX_CALLBACK *funcs)
 void crypto_cleanup_all_ex_data_int(OPENSSL_CTX *ctx)
 {
     int i;
-    OSSL_EX_DATA_GLOBAL *global = openssl_ctx_get_ex_data_global(ctx);
+    OSSL_EX_DATA_GLOBAL *global;
```
Comment (richsalz) [inline crypto/ex_data.c]: again, why split?  please don't make needless changes if the code already follows the style.

File: crypto/ex_data.c (hunk 2 above)
Comment (davxy) [reply crypto/ex_data.c]: I believed it was obsolete. Do you suggest restoring it?

File: crypto/ex_data.c (hunk 1 above)
Comment (davxy) [reply crypto/ex_data.c]: I usually prefer to separate declaration from code involving function calls.

If the community prefer to adopt a single way to perform this, I suggest to put it in the coding guidelines to keep it consistent (sometimes is done in a way, sometimes in the other way round).

File: crypto/ex_data.c (hunk 3 above)
Comment (davxy) [reply crypto/ex_data.c]: See the comment above.

If you want I can put as was before, but (again). Put this in the coding guideline to keep it consistent.

**Comment by davxy** [comment]
Looks like the CI appears to fail for reasons unrelated to the changes.

File: crypto/ex_data.c (hunk 1 above)
Comment (davxy) [reply crypto/ex_data.c]: restored the "inline" initialization

File: crypto/ex_data.c (hunk 2 above)
Comment (richsalz) [reply crypto/ex_data.c]: unless someone else on the project agrees with you, I think it should be restored.

Comment (davxy) [reply crypto/ex_data.c]: Restored

**Comment by slontis** [comment]
travis error is just a timeout that happens quite a bit

**Comment by t8m** [comment]
The failure is just a timeout.

**Comment by t8m** [comment]
Merged to master as 86cd42fbd3
